src/components/EditUserModal.jsx

```diff
@@ -0,0 +1,144 @@
+    props.setShowEditUserModal(false)
+  }
+
+  const handleSubmit = async (event) => {
+    event.preventDefault()
+    await dispatch(updateUserAsync({
+      userId,
+      username,
+      password,
+      bio,
+      email,
+      location
+    })).unwrap()
+
+    handleClose()
+  }
+
+  const clearForm = () => {
+    setUsername('')
+    setPassword('')
+    setBio('')
+    setEmail('')
+    setLocation('')
+  }
+
+  return (
+    <>
+      <Modal
+        show={props.editUserModalOpen}
+        onHide={() => props.setShowEditUserModal(false)}
+        backdrop='static'
+        keyboard={false}
+      >
+        <Modal.Header closeButton>
+          <Modal.Title>Edit Profile</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <Form onSubmit={handleSubmit}>
+            <Form.Group className="mb-3">
+              <Form.Label>Update Username</Form.Label>
+              <Form.Control
+                type="input"
+                placeholder="Enter new username"
+                required
+                autoFocus
+                value={username}
+                onChange={(e) => setUsername(e.target.value)}
+              />
+            </Form.Group>
+            <Form.Group className="mb-3">
+              <Form.Label>Update Password</Form.Label>
+              <Form.Control
+                type="password"
+                placeholder="Enter new password"
+                required
+                autoFocus
+                value={password}
+                onChange={(e) => setPassword(e.target.value)}
+              />
+            </Form.Group>
+            <Form.Group
+              className="mb-3"
+            >
+              <Form.Label>Update Bio</Form.Label>
+              <Form.Control
+                as="textarea"
+                placeholder="Enter something about yourself!"
+                rows={3}
+                value={bio}
+                onChange={(e) => setBio(e.target.value)}
+              />
+            </Form.Group>
+            <Form.Group className="mb-3">
+              <Form.Label>Update Email</Form.Label>
+              <Form.Control
+                type="email"
+                placeholder="Enter new email"
+                required
+                value={email}
+                onChange={e => setEmail(e.target.value)}
+              />
+            </Form.Group>
+            <Form.Label>Update Location</Form.Label>
+            <Form.Select
+              required
+              value={location}
+              onChange={(e) => setLocation(e.target.value)}>
+              <option>Select item location...</option>
+              {locationDropdowns}
+            </Form.Select>
+            <br/>
+            <div className='modal-buttons text-center' >
+              <Button variant="success" type="submit">
+                  Save and Update
+              </Button>{'  '}
+              <Button variant="danger" type="button" onClick={clearForm}>Reset</Button>
```

Comment:
Should the Reset button set everything back to the user's current info instead of clearing the form?, e.g.

```
const clearForm = () => {
    setUsername(currentUser.username)
    setPassword(currentUser.password)
    ...
}
```

Reply:
done. I'll merge to dev tonight